parsl/executors/status_handling.py

```diff
@@ -163,12 +161,8 @@ def handle_errors(self, status: Dict[str, JobStatus]) -> None:
         """
         if not self.block_error_handler:
             return
-        init_blocks = 3
-        if hasattr(self.provider, 'init_blocks'):
-            init_blocks = self.provider.init_blocks
-        if init_blocks < 1:
-            init_blocks = 1
-        error_handler.simple_error_handler(self, status, init_blocks)
+        assert callable(self.block_error_handler)
```

Comment:
with block error handler set to true, as is allowed and should be the default, this assert fails at runtime:

```
jobs.job_status_poller:118 _run_error_handlers INFO: BENC: about to enter handle errors
1691736331.298730 2023-08-11 06:45:31 MainProcess-9203 JobStatusPoller-Timer-Thread-140360055682320-140360045541056 parsl.utils:349 make_callback ERROR: Callback threw an exception - logging and proceeding anyway
Traceback (most recent call last):
  File "/home/benc/parsl/src/parsl/parsl/utils.py", line 347, in make_callback
    self.callback(*self.cb_args)
  File "/home/benc/parsl/src/parsl/parsl/jobs/job_status_poller.py", line 113, in poll
    self._run_error_handlers(self._poll_items)
  File "/home/benc/parsl/src/parsl/parsl/jobs/job_status_poller.py", line 119, in _run_error_handlers
    es.executor.handle_errors(es.status)
  File "/home/benc/parsl/src/parsl/parsl/executors/status_handling.py", line 164, in handle_errors
    assert callable(self.block_error_handler)
AssertionError
```

Reply:
without that assert, mypy statically finds this:

```
parsl/executors/status_handling.py:164: error: "Literal[True]" not callable  [misc]
```

but mypy's interpretation of asserts about types is "I, the programmer, am telling you, mypy, a fact that you cannot figure out for yourself", rather than "I, the programmer, am asking you to statically validate this assert".

Reply:
Fixed.